cli_main.py

```diff
@@ -451,28 +494,14 @@ def main(
 
     with warnings.catch_warnings():
         warnings.simplefilter('always')
-        begin = analysis_kwargs.pop("begin")
-        if begin > u.trajectory.totaltime:
-            raise ValueError("Start ({:.2f} ps) is larer than total time "
-                             "({:.2f} ps).".format(analysis_kwargs["begin"],
-                                                   u.trajectory.totaltime))
-        elif begin > 0:
-            startframe = int(analysis_kwargs["begin"] // u.trajectory.dt)
-        else:
-            startframe = 0
-
-        end = analysis_kwargs.pop("end")
-        if end is not None:
-            stopframe = int(analysis_kwargs["end"] // u.trajectory.dt)
-            analysis_kwargs["end"] += 1  # catch also last frame in loops
-        else:
-            stopframe = None
-
-        dt = analysis_kwargs.pop("dt")
-        if dt > 0:
-            step = int(analysis_kwargs["dt"] // u.trajectory.dt)
-        else:
-            step = 1
+        startframe = convert_str_time(analysis_kwargs.pop("begin"), u.trajectory.dt)
+        if startframe > u.trajectory.n_frames:
```

Comment:
I suggest moving the frame range selection validaity `if startframe > u.trajectory.n_frames` to line `505` after defining `step` with the following, or similar

```python
# raises error if frame selection range is an empty selection
if not list(range(u.trajectory.n_frames)[slice(startframe, stopframe, step)]):
    raise ValueError("Trajectory frame range {}:{}:{} is not valid for total number of frames {}".format(startframe, stopframe, step, u.trajectory.n_frames)
```

Reply:
This is really clever!

Reply:
Thanks!